File: py/2231.py
def isConstructor (n, m):
  sum = m
  m_str = str(m)
  for i in range(0, len(m_str)):
    sum = sum + int(m_str[i])
  return n == sum

n = int(input())
con = 0
ran = n-54 if n>54 else 0

for i in range (n>54, n):
  if isConstructor(n, i):
    con = i
    break

print(con)

File: cpp/2231.cpp
#include <iostream>
#include <string>

using namespace std;

void input (int* i) {
  cin >> *i;
}

void print (int o) {
  cout << o << '\n';
}

bool isConstructor(int n, int m) {
  int sum = m;
  string str = to_string(m);

  for(int i=0; i<str.length(); i++) {
    sum += str[i] - '0';
  }
  
  return n == sum;
}

int main() {
  int n;
  int con = 0;

  input(&n);

  int range = n>54 ? n-54 : 0;

  for (int i=range; i<=n; i++) {
    if (isConstructor(n, i)) {
      con = i;
      break;
    }
  }

  print(con);

  return 0;
}